libs/libcommon/tests/test_simple_cache.py

```diff
@@ -613,3 +614,30 @@ def test_stress_get_cache_reports(num_entries: int) -> None:
         response = get_cache_reports(kind=kind, cursor=next_cursor, limit=100)
         next_cursor = response["next_cursor"]
         assert process_time() - start < MAX_SECONDS
+
+
+def test_get_cache_with_minor_version_for_kind() -> None:
+    kind = "kind"
+    current_version = "2.0.0"
+    minor_version = "1.0.0"
```

Comment:
Yes, it worked with "10.0.0" but it doesn't with "1.0", maybe it is doing some kind of int casting on mongo side, anyway I think we will have to change to a int (Will probably need a migration) or otherwise look for all the cache for a certain `kind` and then, validate the version as in Job_Runner.py but that will imply reading almost all the cache db. What do you think?

Reply:
I think we should move to an integer, because it's useful to be able to use it in mongo queries. We don't really need the details of semver, as we currently just use the major version (x in x.y.z)

Reply:
job_runner_version has been created in https://github.com/huggingface/datasets-server/pull/916 as int, this will fix the str version  issue